### PR TITLE
Feature/feature forums content crud

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewPost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsView.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewForum.java  8
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/groups/Group.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/forums/Forum.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5413408&bid=17748882)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 1]
```diff
@@ -537,12 +565,82 @@ private void setFloatingButtonListener() {
...
+        View newForumDialog = getLayoutInflater().inflate(R.layout.new_forum_dialog, null);
+        dialog.setView(newForumDialog);
+
+        AlertDialog alertDialog = dialog.create();
+        setAddForumButtonListener(alertDialog, newForumDialog);
+
+        alertDialog.show();
+    }
+
+    private void setAddForumButtonListener(AlertDialog alertDialog, View newForumDialog) {
```
Comment (xavier-campos-diaz) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java:628]: You need to comment this method

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+        if (creationDate.getMinutes() < TEN) {
+            forumAuthorDate.append('0');
+        }
+
+        forumAuthorDate.append(creationDate.getMinutes()).append('\n')
+            .append(getResources().getString(R.string.forum_created_by)).append(' ').append(forum.getOwnerUsername());
+    }
+
+    private String getTags() {
```
Comment (xavier-campos-diaz) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java:101]: You should comment this method

**Review by xavier-campos-diaz — COMMENTED** [review]
You should check some issues

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteForum.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewForum.java  8
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostComponentView.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewPost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/groups/Group.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/forums/Forum.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  7
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewPost.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5413408&bid=17748882)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewPost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/forums/Forum.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewForum.java  8
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostComponentView.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  7
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/groups/Group.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewForum.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5413408&bid=17748882)

**Review by xavier-campos-diaz — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/groups/Group.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewForum.java  8
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/forums/Forum.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewForum.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  12
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteForum.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewPost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5413408&bid=17748882)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+     * Method responsible for checking if a user is a subscriber.
+     * @return True if the user is subscriber or false otherwise
+     */
+    private boolean isUserSubscriber() {
+        return forum.getGroup().isUserSubscriber(InfoGroupFragment.getCommunication().getUser());
+    }
+
+    /**
+     * Method responsible for checking if a message is empty.
+     * @return True if the message is empty or false otherwise
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java:73]: Missing param tag

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularEntryView.java [hunk 4]
```diff
@@ -5,28 +5,31 @@
...
 import androidx.constraintlayout.widget.ConstraintLayout;
 import androidx.constraintlayout.widget.ConstraintSet;
 
 import org.pesmypetcare.mypetcare.R;
 import org.pesmypetcare.mypetcare.features.pets.Pet;
 
 public abstract class CircularEntryView extends ConstraintLayout {
-    private Context currentActivity;
-    private Pet pet;
     private final int PADDING = 15;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CircularEntryView.java]: missing static modifier

**Review by santidrj — CHANGES_REQUESTED** [review]
Need some changes

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsComponentView.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewForum.java  8
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  12
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddNewPost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/ForumsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/groups/Group.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PostComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteForum.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/community/forums/Forum.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteForum.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddNewPost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePost.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5413408&bid=17748882)

**Review by santidrj — APPROVED** [review]
👍

**Pull request merged**